main.go

```diff
@@ -216,7 +216,7 @@ func main() {
 		RetryPeriod:                &retryPeriod,
 		LeaderElectionID:           config.LeaderElectionKey,
 		LeaderElectionNamespace:    config.LeaderElectionNamespace,
-		LeaderElectionResourceLock: resourcelock.ConfigMapsResourceLock,
```

Comment:
we'd also need rbac permissions for lease resources. 
config/rbac/leader_election_role.yaml and the installation manifests

Reply:
@kishorj, I have updated. Please take a look.

Reply:
I have tested this change and it looks fine to me. Can you do the following changes
```
- apiGroups:
  - coordination.k8s.io
  resources:
  - leases
  verbs:
  - create
- apiGroups:
  - coordination.k8s.io
  resourceNames:
  - cp-vpc-resource-controller
  resources:
  - leases
  verbs:
  - get
  - update
```
in RBAC

and adding the annotation in main
```
// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=create
// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,namespace=kube-system,resourceNames=cp-vpc-resource-controller,verbs=get;update
```

Actually the annotation will auto add the permissions into config/rbac/role.yaml when building. Thanks.